zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -177,14 +185,21 @@ def working_set(
     links=(),
     index=None,
     allow_hosts=('*',),
+    cache_storage=_global_working_set_cache,
 ):
     """Helper function to build a working set.
 
     Return an instance of `pkg_resources.WorkingSet`.
 
-    Results are cached in the `_working_set_cache` dict at module level.
+    Results are cached in the mapping provided in the `cache_storage` argument.
+    This defaults to a global dict at this module. If `None` is passed then
+    caching is disabled.
```

Comment:
> The reason is I think the working_set() function at module level is useful on its own.
> if people choose to use it then it's nice to have caching enabled by default, since building working sets is slow.

> I think it's better to do this than putting the burden of setting up the cache storage in the callers.

Personally, I happen to disagree, I think module level caches are dangerous by default. Moreover...:

> I think it would be nice to use the working_set() function here, instead of instantiating the Egg class.

That seems to disprove the point. That recipe *already* has an Egg instance, which has the buildout, etc, and so access to the scoped cache. That's been the only way to do it so far, and I don't see a reason to introduce another way to do it ("There should be one-- and preferably only one --obvious way to do it."), especially if that way is potentially dangerous.